decode.go

```diff
@@ -26,31 +26,36 @@ type codecConverter struct {
 	converter NodeConverter
 }
 
-var codecTable = map[uint64]codecConverter{}
-
-// RegisterCodec registers a specialized prototype & converter for a specific codec
-func RegisterCodec(codec uint64, prototype ipld.NodePrototype, converter NodeConverter) {
-	codecTable[codec] = codecConverter{prototype, converter}
+type Decoder struct {
+	codecTable     map[uint64]codecConverter
+	linkSystemBase ipld.LinkSystem
 }
 
-var linkSystemBase ipld.LinkSystem
+func NewDecoder() *Decoder {
```

Comment:
@rvagg @willscott WDYT about having this take a linksystem (or at least the relevant parts for building a linksystem like the decoder and reifier) instead of being stuck with the go-ipld-prime default registry? Could be a functional option if you'd prefer `NewDecoder()` to continue to work.

Reply:
Is there a different linksystem you need for functionality beyond the transformations exposed by the codec registry here?

would a `Decoder.SetLinkSystem(ls)` method work?

Reply:
Well it's not just the transformations it's also the passthrough for non-transformed codecs to the go-ipld-prime defaults. I'd like to control those instead of being reliant on whatever gets autoloaded by some package I imported.

Sure a setter would work too. It seems mostly like a stylistic choice compared to a constructor argument so 🤷‍♀️

Reply:
adding a method isn't a breaking interface change, so would be v0.2.1 - i guess it doesn't matter too much :)

Reply:
https://github.com/ipfs/go-ipld-legacy/pull/14